src/mpyl/utilities/docker/__init__.py

```diff
@@ -184,7 +186,11 @@ def push_to_registry(
     image = docker.image.inspect(image_name)
     logger.debug(f"Found image {image}")
 
-    login(logger=logger, registry_config=docker_config)
+    if docker_config.provider == "aws":
+        login_ecr(logger=logger, registry_config=docker_config)
+    if docker_config.provider == "azure":
```

Comment:
I would move these `if` checks into the `login` function itself. that prevents duplicated this piece of code, maybe also add an `else` with some logging if a provider is passed that doesn't have a login (yet)?

Reply:
Merged both login functions into one.